src/hypothesis/extra/datetime.py

```diff
@@ -123,3 +123,34 @@ def times(allow_naive=None, timezones=None):
 
 def datetime_to_time(dt):
     return dt.timetz()
+
+
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=dt.timedelta.min, max_value=dt.timedelta.max):
+        assert type(min_value) == dt.timedelta, 'min_value must be a timedelta'
+        assert type(max_value) == dt.timedelta, 'min_value must be a timedelta'
+        assert min_value <= max_value,\
+            'max_value must not be smaller than min_value'
+
+        SECS_IN_DAY = 3600 * 24
+        MICROS_IN_SEC = 1000000
+
+        max_days, min_days = max_value.days, min_value.days
+        max_secs, min_secs = max_value.seconds, min_value.seconds
+        max_ms, min_ms = max_value.microseconds, min_value.microseconds
+
+        self.max_micros = (max_ms + (max_secs * MICROS_IN_SEC) +
+                           (max_days * SECS_IN_DAY * MICROS_IN_SEC))
+        self.min_micros = (min_ms + (min_secs * MICROS_IN_SEC) +
+                           (min_days * SECS_IN_DAY * MICROS_IN_SEC))
```

Comment:
I’m mildly annoyed that the `timedelta` object doesn’t expose a `total_microseconds()` method (there’s a `total_seconds()` method, but it’s not precise enough for our purposes).

Worth commonising this into a function that’s internal to this function? Maybe not, just a thought.